src/server/JasmineGraphInstanceService.cpp

```diff
@@ -584,6 +587,271 @@ void *instanceservicesession(void *dummyPt) {
+            instance_logger.log("Received Worker List " + workerList, "info");
+
+            std::vector<string> workerSockets;
+            boost::split(workerSockets, workerList, boost::is_any_of(","));
+
+            // Calculate the out degree distribution in the current super worker.
+            JasmineGraphHashMapLocalStore graphDB;
+            JasmineGraphHashMapCentralStore centralDB;
+            std::map<std::string, JasmineGraphHashMapLocalStore>::iterator it;
+            std::map<std::string, JasmineGraphHashMapCentralStore>::iterator itcen;
+
+            if (JasmineGraphInstanceService::isGraphDBExists(graphID, partitionID)) {
+                JasmineGraphInstanceService::loadLocalStore(graphID, partitionID, graphDBMapLocalStores);
+            }
+
+            if (JasmineGraphInstanceService::isInstanceCentralStoreExists(graphID, partitionID)) {
+                JasmineGraphInstanceService::loadInstanceCentralStore(graphID, partitionID, graphDBMapCentralStores);
+            }
+
+            graphDB = graphDBMapLocalStores[graphID + "_" + partitionID];
+            centralDB = graphDBMapCentralStores[graphID + "_centralstore_" + partitionID];
+
+            map<long,long> degreeDistribution = graphDB.getInDegreeDistributionHashMap();
+            std::map<long,long>::iterator its;
+
+            map<long,long> degreeDistributionCentral = centralDB.getInDegreeDistributionHashMap();
+            std::map<long,long>::iterator itcentral;
+
+            for (its = degreeDistributionCentral.begin(); its != degreeDistributionCentral.end();++its) {
+
+                bool centralNodeFound = false;
+                for (itcentral = degreeDistribution.begin(); itcentral != degreeDistribution.end();++itcentral) {
+
+                    if ((its->first) == (itcentral->first)) {
+                        degreeDistribution[its->first] = (its->second) + (itcentral->second);
+                        centralNodeFound = true;
+                    }
+                }
+                if (!centralNodeFound) {
+                    degreeDistribution.insert(std::make_pair(its->first, its->second));
+                }
+            }
+
+            // Invoke other workers to calculate their own our degree distributions
+            //todo  invoke other workers asynchronously
+            for (vector<string>::iterator workerIt=workerSockets.begin(); workerIt!=workerSockets.end(); ++workerIt) {
+                instance_logger.log("Worker pair " + *workerIt, "info");
+
+                std::vector <string> workerSocketPair;
+                boost::split(workerSocketPair, *workerIt, boost::is_any_of(":"));
+
+                if  (std::to_string(serverPort).compare(workerSocketPair[1]) == 0) {
+                    continue;
+                }
+
+                string host = workerSocketPair[0];
+                int port = stoi(workerSocketPair[1]);
+                int sockfd;
+                char data[300];
```

Comment:
Can we have a constant here rather than `300`?

Reply:
Fixed

Reply:
Done